src/commands/registration/stor.js

```diff
@@ -37,23 +37,30 @@ module.exports = {
       });
 
       const socketPromise = new Promise((resolve, reject) => {
-        this.connector.socket.on('data', (data) => {
-          if (this.connector.socket) this.connector.socket.pause();
-          if (stream && stream.writable) {
-            stream.write(data, () => this.connector.socket && this.connector.socket.resume());
-          }
-        });
-        this.connector.socket.once('end', () => {
-          if (stream.listenerCount('close')) stream.emit('close');
-          else stream.end();
-          resolve();
-        });
+        if (pipeAllowed) {
+          this.connector.socket.pipe(stream);
+        } else { 
+          this.connector.socket.on('data', (data) => {
+            if (this.connector.socket) this.connector.socket.pause();
+            if (stream && stream.writable) {
+              stream.write(data, () => { 
+                this.connector.socket && this.connector.socket.resume();
+              });
+            }
+          });
+       
+          this.connector.socket.once('end', () => {
+            if (stream.listenerCount('close')) stream.emit('close');
+            else stream.end();
+            resolve();
+          });
+        }
```

Comment:
We might actually be able to do this without requiring a new file system setting (`pipeAllowed`). We can replace the `data` event, since really all this is doing is re-implementing `pipe` poorly. By specifying the `end` option we can still handle the `close` event on the given stream if set.

```suggestion
        this.connector.socket.pipe(stream, {end: false});
        this.connector.socket.once('end', () => {
          if (stream.listenerCount('close')) stream.emit('close');
          else stream.end();
          resolve();
        });
```

Reply:
Perfect implementation, my intention was not to touch the latter solution, 👍🏽